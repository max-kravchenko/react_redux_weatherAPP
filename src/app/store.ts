import { configureStore, combineReducers } from '@reduxjs/toolkit';
import weatherSliceReducer from '../features/weatherSlice';

const rootReducer = combineReducers({
  weatherSliceReducer,
});

export const store = configureStore({
  reducer: rootReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = typeof store; 
export type AppDispatch = AppStore['dispatch'];
