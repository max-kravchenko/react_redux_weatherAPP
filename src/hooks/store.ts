import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { AppDispatch, RootState } from '../app/store';

export const useCustomDispatch = () => useDispatch<AppDispatch>();

export const useCustomSelector: TypedUseSelectorHook<RootState> = useSelector;