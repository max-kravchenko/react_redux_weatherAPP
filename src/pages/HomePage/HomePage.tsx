import React, { useEffect, useState } from 'react';
import { SearchForm } from '../../components/SearchForm';
import { WeatherList } from '../../components/WeatherList';
import { useCustomDispatch, useCustomSelector } from '../../hooks/store';
import { selectCurrentWeatherData } from '../../selectors/selector';
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';
import { Typography } from '@mui/material';
import { weatherSlice } from '../../features/weatherSlice';

export const HomePage = React.memo(function HomePage() {
  const [city, setCity] = useState<string>('');

  const dispatch = useCustomDispatch();

  const { weather } = useCustomSelector(selectCurrentWeatherData);

  useEffect(() => {
    dispatch(weatherSlice.actions.set(weather));
  }, []);

  return (
    <Container>
      <Box 
        display="flex"
        justifyContent="center"
        alignItems="center"
      >
        <Typography gutterBottom variant="h2">
      Weather App
        </Typography>
      </Box>

      <Box 
        display="flex"
        justifyContent="center"
        alignItems="center"
        paddingBottom={4}
      >
        <SearchForm city={city} setCity={setCity} />
      </Box>
      <WeatherList cities={weather} />
    </Container>
  );
});