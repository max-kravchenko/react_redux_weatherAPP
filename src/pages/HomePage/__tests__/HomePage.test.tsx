import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event/';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { HomePage } from '../HomePage';

describe('Homepage', () => {
  it('should enter value correctly', async () => {
    render(<Provider store={store}> <HomePage/> </Provider>);

    expect(screen.queryByText(/Rivne/)).toBeNull();

    await userEvent.type(screen.getByRole('textbox'), 'Rivne');

    expect(
      screen.getByRole('textbox').value).toBe('Rivne');
  });

  it('should render page correctly', () => {
    expect(render(<Provider store={store}> <HomePage/> </Provider>)).toMatchSnapshot();
  });
});
