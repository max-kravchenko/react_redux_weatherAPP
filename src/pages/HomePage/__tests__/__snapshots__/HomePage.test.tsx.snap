// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-1n2mv2k"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
          >
            Weather App
          </h2>
        </div>
        <div
          class="MuiBox-root css-901fzl"
        >
          <div
            class="MuiBox-root css-0"
          >
            <form
              autocomplete="off"
              class="MuiBox-root css-1n2mv2k"
              novalidate=""
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="content-input"
                    id=":r1:"
                    placeholder="Choose a city"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeLarge MuiButton-textSizeLarge css-xymarw-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Show weather
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall css-6ekzlo-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Clear selections
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </form>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
        />
      </div>
       
    </div>
  </body>,
  "container": <div>
     
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-1n2mv2k"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
        >
          Weather App
        </h2>
      </div>
      <div
        class="MuiBox-root css-901fzl"
      >
        <div
          class="MuiBox-root css-0"
        >
          <form
            autocomplete="off"
            class="MuiBox-root css-1n2mv2k"
            novalidate=""
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="content-input"
                  id=":r1:"
                  placeholder="Choose a city"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textSuccess MuiButton-sizeLarge MuiButton-textSizeLarge css-xymarw-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Show weather
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall css-6ekzlo-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Clear selections
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-480o17-MuiGrid-root"
      />
    </div>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
